validator_client/src/block_service.rs

```diff
@@ -357,59 +367,212 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
                                     "block_slot" => ?slot,
                                     "info" => "blinded proposal failed, attempting full block"
```

Comment:
If I understand correctly, `BlockError::Recoverable` and this error log probably don't apply here since there's no fallback attempt on the VC side when using the block v3 endpoint?

Reply:
Yes, I believe there's no need for a fallback attempt in the VC as the block v3 endpoint handles the fallback logic on its own. However the old blinded v2 endpoint also had similar fallback logic, it would just return a blinded payload from the local execution engine if the builder failed for any reason. So I'm not really sure why the fallback mechanism in the VC existed in the first place.

Reply:
Ah I got what you mean now. I think it the VC fallback was implemented first and then improved by implementing the fallback behaviour in the beacon node, so VC doesn't have to make multiple calls to the BN. It's explained here:
> ..adding pre-bellatrix block support on blinded endpoints and allowing the normal proposal flow (local payload construction) on blinded endpoints. This resulted in better fallback logic because the VC will not have to switch endpoints on failure in the BN <> Builder API, the BN can just fallback immediately and without repeating block processing that it shouldn't need to. We can also keep VC fallback from the VC<>BN API's blinded endpoint to full endpoint.

source: https://github.com/sigp/lighthouse/pull/3134

However the fallback behaviour in VC may still be useful for users using a beacon node that doesn't have fallback behaviour built in, so that's probably why it remained here. I think we can keep the VC fallback logic when using v2 but it won't be applicable to the v3 endpoint I believe.

Reply:
Actually I wonder if it would be useful to fallback to the v2 endpoint, so proposer doesn't miss the block if there's an issue on the new v3 endpoint?